src/lib.rs

```diff
@@ -1475,6 +1478,122 @@ where
         .join(" ")
 }
 
+fn serialize_with_join_vec<T, S>(value: &[T], serializer: S) -> Result<S::Ok, S::Error>
+where
+    T: AsRef<str>,
+    S: Serializer,
+{
+    serializer.serialize_str(&join_vec(value))
+}
+
+///
+/// A request to the end session endpoint.
+///
+pub struct LogoutRequest {
+    end_session_endpoint: EndSessionUrl,
+    parameters: LogoutRequestParameters,
+}
+
+#[derive(Serialize, Default)]
+struct LogoutRequestParameters {
+    id_token_hint: Option<String>,
+    logout_hint: Option<LogoutHint>,
+    client_id: Option<ClientId>,
+    post_logout_redirect_uri: Option<RedirectUrl>,
+    state: Option<CsrfToken>,
+    #[serde(serialize_with = "serialize_with_join_vec")]
+    ui_locales: Vec<LanguageTag>,
+}
+
+impl From<EndSessionUrl> for LogoutRequest {
+    fn from(value: EndSessionUrl) -> Self {
+        LogoutRequest {
+            end_session_endpoint: value,
+            parameters: Default::default(),
+        }
+    }
+}
+
+impl LogoutRequest {
+    ///
+    /// Provides an ID token previously issued by this OpenID Connect Provider as a hint about
+    /// the user's identity.
+    ///
+    pub fn set_id_token_hint<AC, GC, JE, JS, JT>(
+        mut self,
+        id_token_hint: &IdToken<AC, GC, JE, JS, JT>,
+    ) -> Self
+    where
+        AC: AdditionalClaims,
+        GC: GenderClaim,
+        JE: JweContentEncryptionAlgorithm<JT>,
+        JS: JwsSigningAlgorithm<JT>,
+        JT: JsonWebKeyType,
+    {
+        self.parameters.id_token_hint = Some(id_token_hint.to_string());
+        self
+    }
+
+    ///
+    /// Provides the OpenID Connect Provider with a hint about the user's identity.
+    ///
+    /// The nature of this hint is specific to each provider.
+    ///
+    pub fn set_logout_hint(mut self, logout_hint: LogoutHint) -> Self {
+        self.parameters.logout_hint = Some(logout_hint);
+        self
+    }
+
+    ///
+    /// Provides the OpenID Connect Provider with the client identifier.
+    ///
+    /// When both this and `id_token_hint` are set, the provider must verify that
+    /// this client id matches the one used when the ID token was issued.
+    ///
+    pub fn set_client_id(mut self, client_id: ClientId) -> Self {
+        self.parameters.client_id = Some(client_id);
+        self
+    }
+
+    ///
+    /// Provides the OpenID Connect Provider with a URI to redirect to after
+    /// the logout has been performed.
+    ///
+    pub fn set_redirect_uri(mut self, redirect_uri: RedirectUrl) -> Self {
```

Comment:
Let's call this `set_post_logout_redirect_uri`. Also, the `RedirectUrl` type is intended to refer to the authorization endpoint's redirect URI. Since these redirect URLs serve different purposes and are used for different endpoints, let's define a separate `PostLogoutRedirectUrl` type.